eng/common/AzurePipelinesTemplates/Maestro-EnforceDependencyMatching-Steps.yml

```diff
@@ -0,0 +1,108 @@
+
+- task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2
+  displayName: RestoreNuGetPackages
+  inputs:
+    restoreSolution: eng/common/packages.config
+    feedsToUse: config
+    nugetConfigPath: eng/common/nuget.config
+    restoreDirectory: packages
+
+- task: PowerShell@2
+  displayName: Extract Nuspec and Enforce versions 
+  inputs:
+    targetType: 'inline'
+    script: |
+      $enforcementSet = @{}
+      '${{ parameters.EnforcementList }}'.split([Environment]::NewLine) | ForEach {
+        if ($_ -ne '')
+        {
+          $enforcementSet[$_] = $true
+          Write-Host "Enforcing $_"
+        }
+      }
+
+      [xml]$versionDetails = Get-Content -Path '$(Build.SourcesDirectory)\eng\Version.Details.xml'
+      $directory = "$(Build.SourcesDirectory)/eng/common/packages"
+
+      # Set up Map from the dependencies to its versions to reference later
+      $dependenciesMap = @{}
+      foreach ($dependency in $versionDetails.Dependencies.ProductDependencies.Dependency)
+      {
+        $id = $dependency.name
+        $version = $dependency.version
+        $dependenciesMap[$id] = $version
+      }
+
+      $seen = @{}
+      $files = Get-ChildItem $directory
+      foreach ($file in $files) # Iterate through each package we restored in the directory
+      {
+        Write-Host $file.FullName
+        $nupkgPaths = Get-ChildItem $file.FullName -Filter "*.nupkg"
+
+        # Extract nupkg to access the nuspec
+        # The files in this directory does not contain the nuspec by default
+        foreach ($nupkgPath in $nupkgPaths)
+        {
+          $rename = $nupkgPath.Name + ".zip"
+          Rename-Item $nupkgPath.FullName $rename
+          $renamedFilePath = $nupkgPath.FullName + ".zip"
+          $dest = $file.FullName + "/contents"
+          Expand-Archive $renamedFilePath -Destination $dest
+        }
+        
+        $nuspecPaths = Get-ChildItem $file.FullName -Recurse -Filter "*.nuspec"
+        foreach ($nuspecPath in $nuspecPaths)
+        {
+          # When Nuget restore the packages, the packages directory can have
+          # two of the same package in the directory, one with the version,
+          # and one without. This "seen" set prevents the scripts from looking at it twice
+          if (-not $seen.Contains($nuspecPath.Name.ToLower()))
+          {
+            $seen[$nuspecPath.Name.ToLower()] = $true
+            Write-Host "Found Nuspecs"
+            Write-Host "Comparing dependencies in " $nuspecPath.Name.ToLower()
+            [xml]$nuspec = Get-Content -Path $nuspecPath.FullName
+            foreach ($group in $nuspec.package.metadata.dependencies.group)
+            {
+              foreach ($dependency in $group.dependency)
+              {
+                $id = $dependency.id
+                # WinUI encapsulated their versions with brackets so we have to sanitize it
```

Comment:
If I'm reading right, this will do the right thing if they fix / align with the other package. Yes?